### ISSUE TITLE
Manually deleting Web App and running terraform apply throws a 404 ResourceNotFound

### ISSUE THREAD

**Issue by gettek** [terraform-providers/terraform-provider-azurerm#3300]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.11.13

### Affected Resource(s)

* `azurerm_v1.25.0`

### Terraform Configuration Files

```hcl
 terraform {
  required_version = ">= 0.11" 
  backend "azurerm" {
    resource_group_name  = "__tfstoragerg__"
    storage_account_name = "__tfstorageaccount__"
    container_name       = "states"
    key                  = "__productName__.tfstate"
	  access_key           = "__tfstoragekey__"
	}
}

resource "azurerm_resource_group" "rg" {
  name     = "${var.product}${var.env}"
  location = "${var.env_vars["location"]}"

  tags {
    AutoShutdownSchedule = "N/A"
    BusinessUnit         = "${var.bu}"
    ContactDetails       = "N/A"
    CostCentre           = "${var.bu}"
    DataClassification   = "TBC"
    ExpirationDate       = "TBC"
    Environment          = "${var.env}"
    MaintenanceWindows   = "TBC"
    OperatingSystem      = "N/A"
    ProjectCode          = "${var.product}${var.env}"
    SecurityGrade        = "TBC"
    ServiceName          = "${var.product}${var.env}"
    ServiceTier          = "Silver"
  }
}

# Container Registry
data "azurerm_container_registry" "acr" {
  name                = "acr"
  resource_group_name = "acr"
}

# SQL Server
resource "azurerm_sql_server" "sql" {
  name                         = "${var.product_prefix}${var.env}"
  resource_group_name          = "${azurerm_resource_group.rg.name}"
  location                     = "${azurerm_resource_group.rg.location}"
  version                      = "${var.sql["version"]}"
  administrator_login          = "${var.sql["admin_usr"]}"
  administrator_login_password = "${var.sql["admin_pwd"]}"
}

# SQL DB
resource "azurerm_sql_database" "db" {
  name                = "${var.product_prefix}${var.env}db"
  resource_group_name = "${azurerm_resource_group.rg.name}"
  location            = "${azurerm_resource_group.rg.location}"
  server_name         = "${azurerm_sql_server.sql.name}"
  collation           = "${var.sql["collation"]}"
  edition             = "${var.sql["edition"]}"
  max_size_bytes      = "${var.sql["max_size_bytes"]}"
}

# SQL Firewall Rules
resource "azurerm_sql_firewall_rule" "sqlfw" {
  name                = "AllowAllWindowsAzureIps"
  resource_group_name = "${azurerm_resource_group.rg.name}"
  server_name         = "${azurerm_sql_server.sql.name}"
  start_ip_address    = "0.0.0.0"
  end_ip_address      = "0.0.0.0"
}

# App Service Plan
resource "azurerm_app_service_plan" "asp" {
  name                = "${var.product_prefix}${var.env}"
  location            = "${azurerm_resource_group.rg.location}"
  resource_group_name = "${azurerm_resource_group.rg.name}"
  kind                = "Linux"
  reserved            = true

  sku {
    tier     = "Standard"
    size     = "S1"
    capacity = "1"
  }
}

# App Service
resource "azurerm_app_service" "app" {
  name                = "${var.product_prefix}${var.env}"
  location            = "${azurerm_resource_group.rg.location}"
  resource_group_name = "${azurerm_resource_group.rg.name}"
  app_service_plan_id = "${azurerm_app_service_plan.asp.id}"

  site_config {
    app_command_line         = ""
    linux_fx_version         = "DOCKER|appsvc/sample-hello-world:latest"
  }
  lifecycle {
    ignore_changes = [
      "site_config.0.linux_fx_version", # deployments are made outside of Terraform
    ]
  }

  app_settings = {
    "DOCKER_REGISTRY_SERVER_URL"          = "https://${data.azurerm_container_registry.acr.login_server}"
    "DOCKER_REGISTRY_SERVER_USERNAME"     = "${data.azurerm_container_registry.acr.admin_username}"
    "DOCKER_REGISTRY_SERVER_PASSWORD"     = "${data.azurerm_container_registry.acr.admin_password}"
    "WEBSITES_ENABLE_APP_SERVICE_STORAGE" = "false"
  }

  connection_string {
    name  = "defaultConnection"
    type  = "SQLAzure"
    value = "Data Source=tcp:${azurerm_sql_server.sql.fully_qualified_domain_name},1433;Initial Catalog=${azurerm_sql_database.db.name};User Id=${var.sql["admin_usr"]};Password=${var.sql["admin_pwd"]};"
  }
}
```

### Expected Behavior

`terraform apply` should generate a create plan for the function app

### Actual Behavior

When running `terraform apply` the following error is thrown during the refresh state step.
Error refreshing state.
```log
* azurerm_app_service.app: azurerm_app_service.app: Error making Read request on AzureRM App Service Configuration "XXXXXX": web.AppsClient#GetConfiguration: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Web/sites/XXXXXX' under resource group 'XXXXXXX' was not found."
```

### Steps to Reproduce

1. `terraform apply`
2. Manually delete the Web App Service via Azure portal.
3. `terraform apply`

### Important Factoids

Provisioning succeeds if effected resources are removed from state file

### References

* #1722
* #2704

**Comment by gettek** [comment]
Just noticed possible fix in [v1.25.0](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.25.0/azurerm/resource_arm_app_service.go#L436)

Running checks now

**Comment by gettek** [comment]
Fix Confirmed, Thanks!
